src/validation.c

```diff
@@ -331,6 +339,24 @@ validate_netdef_grammar(const NetplanParser* npp, NetplanNetDefinition* nd, yaml
             return yaml_error(npp, node, error, "%s: invalid id '%u' (allowed values are 0 to 4094)", nd->id, nd->vlan_id);
     }
 
+    if (nd->type == NETPLAN_DEF_TYPE_TUNNEL &&
+        nd->tunnel.mode == NETPLAN_TUNNEL_MODE_VXLAN) {
+        if (nd->link)
+            nd->link->has_vxlans = TRUE;
+        else
+            nd->vxlan.independent = TRUE;
```

Comment:
nitpick: I'm not wild about mutating a netdef in the validation phase. Couldn't this be done at the end of the vxlan parsing instead?

Reply:
ACK! Moving it into `parse.c`, where we now have a "Preprocessing" and "Postprocessing" stage directly before/after running the individual settings handlers.